generative_ai/embeddings/document_retrieval_example.py

```diff
@@ -19,27 +19,41 @@
 
 
 def embed_text() -> list[list[float]]:
-    """Embeds texts with a pre-trained, foundational model.
+    """Calculates embeddings for the given text inputs.
 
     Returns:
-        A list of lists containing the embedding vectors for each input text
+        A list of lists containing the embedding vectors for each input text.
     """
 
-    # A list of texts to be embedded.
-    texts = ["banana muffins? ", "banana bread? banana muffins?"]
+    model = TextEmbeddingModel.from_pretrained("text-embedding-004")
+
+    inputs = [
+        # Option1: Use TextEmbeddingInput object as embedding input
+        TextEmbeddingInput(
+            text=(
+                "You'll find some apps on your Home screens, and all your apps in All Apps. "
+                "You can open apps, switch between apps, and find 2 apps at once."
+            ),
+            title="Find, open & close apps on a Pixel phone",
+            # The task type for embedding. Check the available tasks in the model's documentation.
+            task_type="RETRIEVAL_DOCUMENT"
+        ),
+        # Option2: Use plain text as embedding input
+        (
+            "You can download no-charge and paid apps from Google Play on your Android phone or tablet. "
+            "We recommend that you get apps from Google Play, but you can also get them from other sources."
+        )
+    ]
+
     # The dimensionality of the output embeddings.
-    dimensionality = 256
-    # The task type for embedding. Check the available tasks in the model's documentation.
-    task = "RETRIEVAL_DOCUMENT"
+    dimensionality = 128
+    embeddings = model.get_embeddings(inputs, output_dimensionality=dimensionality)
 
-    model = TextEmbeddingModel.from_pretrained("text-embedding-004")
-    inputs = [TextEmbeddingInput(text, task) for text in texts]
-    kwargs = dict(output_dimensionality=dimensionality) if dimensionality else {}
```

Comment:
This line us utterly confusing. We create a `dict` with just one argument, only to expand it with `**` on the next line. The `dimensionality` is always defined, so there is no need for `else` 🤷

Reply:
I believe, the idea was to share info `kwargs` option.

Reply:
Unless it was not a hinderance or deprecated, keep it for now. At a later a time, we will re-review all code samples.

Reply:
> I believe, the idea was to share info kwargs option.

@msampathkumar , this comment is not clear to me. We're passing `**kwargs` in the line below, which is an equivalent of passing named arguments but in a more obscure way. I believe that this sample does not foster Python best practices and it's worth fixing it.